libs/vkd3d/resource.c

```diff
@@ -4597,6 +4597,12 @@ void d3d12_desc_create_cbv_embedded(vkd3d_cpu_descriptor_va_t desc_va,
     addr_info.format = VK_FORMAT_UNDEFINED;
     addr_info.address = desc->BufferLocation;
     addr_info.range = desc->SizeInBytes;
+
+    /* For robustness purposes. If someone tries to access a UBO as an image,
+     * it should translate to a NULL descriptor. */
+    d3d12_descriptor_heap_write_null_descriptor_template_embedded_partial(device, desc_va,
+            VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 0, device->bindless_state.descriptor_buffer_packed_raw_buffer_offset);
```

Comment:
Is `VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER` correct here when the goal is to null out the image portion of the descriptor?

Reply:
Added some comments to make it a bit more clear why it's done the way it's done.